### PR TITLE
Add specification ID to test case names

### PR DESCRIPTION
**Pull request by thekenu**
### Summary

Where applicable, add  specification ID to test case names

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/DIM/Test/Src/Test_StateMachine.cpp [hunk 1]
```diff
@@ -179,7 +179,7 @@ class DimStateMachineTest : public testing::Test
     struct RotarySwitch *     drive_mode_switch;
 };
 
-TEST_F(DimStateMachineTest, check_drive_state_is_broadcasted_over_can)
+TEST_F(DimStateMachineTest, check_drive_state_is_broadcasted_over_can_DIM_12)
```
Comment (thekenu) [inline boards/DIM/Test/Src/Test_StateMachine.cpp:182]: I thought about leaving the ID as a comment, but it seems more concise to just leave the ID in the test case name

File: boards/PDM/Test/Src/Test_StateMachine.cpp [hunk 2]
```diff
@@ -390,7 +396,9 @@ TEST_F(PdmStateMachineTest, check_24v_acc_voltage_can_signals_in_all_states)
         CANMSGS_PDM_ERRORS__24_V_ACC_VOLTAGE_OUT_OF_RANGE_OVERFLOW_CHOICE);
 }
 
-TEST_F(PdmStateMachineTest, check_aux1_current_can_signals_in_non_init_states)
+TEST_F(
+    PdmStateMachineTest,
+    check_aux1_current_can_signals_in_non_init_states_PDM_11_13)
```
Comment (thekenu) [inline boards/PDM/Test/Src/Test_StateMachine.cpp:401]: This means this test applies to two IDs - 11 and 13

File: boards/BMS/Test/Src/Test_StateMachine.cpp [hunk 3]
```diff
@@ -113,7 +113,7 @@ class BmsStateMachineTest : public testing::Test
     struct RgbLedSequence *   rgb_led_sequence;
 };
 
-TEST_F(BmsStateMachineTest, check_init_state_is_broadcasted_over_can)
+TEST_F(BmsStateMachineTest, check_init_state_is_broadcasted_over_can_BMS_31)
```
Comment (garethellis0) [inline boards/BMS/Test/Src/Test_StateMachine.cpp:116]: What do you think about `BMS_31__check_init_state_is_broadcasted_over_can` or `check_init_state_is_broadcasted_over_can__BMS_31`? Just trying to make it a bit more clear that `BMS_31` is somewhat seperate.

File: boards/PDM/Test/Src/Test_StateMachine.cpp (hunk 2 above)
Comment (garethellis0) [reply boards/PDM/Test/Src/Test_StateMachine.cpp:401]: I think I'd prefer `PDM_11_to_13`, just to be super explicit?

File: boards/DIM/Test/Src/Test_StateMachine.cpp (hunk 1 above)
Comment (garethellis0) [reply boards/DIM/Test/Src/Test_StateMachine.cpp:182]: Either one is fine by me.

File: boards/DIM/Test/Src/Test_StateMachine.cpp [hunk 4]
```diff
@@ -244,7 +244,46 @@ TEST_F(
...
+    App_SharedStateMachine_Tick(state_machine);
+    ASSERT_EQ(
+        100,
+        App_CanTx_GetPeriodicSignal_MAPPED_PADDLE_POSITION(can_tx_interface));
+    get_raw_paddle_position_fake.return_val = 96;
+    App_SharedStateMachine_Tick(state_machine);
+    ASSERT_EQ(
+        100,
+        App_CanTx_GetPeriodicSignal_MAPPED_PADDLE_POSITION(can_tx_interface));
+}
```
Comment (garethellis0) [inline boards/DIM/Test/Src/Test_StateMachine.cpp:282]: This doesn't seem like part of this PR?

File: boards/BMS/Test/Src/Test_StateMachine.cpp (hunk 3 above)
Comment (thekenu) [reply boards/BMS/Test/Src/Test_StateMachine.cpp:116]: I think that's a good point, let me add the double underscore.

File: boards/DIM/Test/Src/Test_StateMachine.cpp (hunk 4 above)
Comment (thekenu) [reply boards/DIM/Test/Src/Test_StateMachine.cpp:282]: Ah yes forgot to leave a comment but as i was adding the specification ID I realized we forgot to write a test case for this particular ID. Slightly out-of-scope but I just quickly wrote a test for it.

File: boards/PDM/Test/Src/Test_StateMachine.cpp (hunk 2 above)
Comment (thekenu) [reply boards/PDM/Test/Src/Test_StateMachine.cpp:401]: Yeah that's a good point i'll fix that

File: boards/DIM/Test/Src/Test_StateMachine.cpp (hunk 1 above)
Comment (thekenu) [reply boards/DIM/Test/Src/Test_StateMachine.cpp:182]: Actually after considering the ambiguity you pointed out (e.g. single vs double underscore, misinterpreting multiple IDs as a contiguous range of IDs), I think it's better to leave things as comments.

**Comment by thekenu** [comment]
@garethellis0 Ready for another review :octocat:

File: boards/DIM/Test/Src/Test_StateMachine.cpp [hunk 5]
```diff
@@ -188,6 +189,7 @@ TEST_F(DimStateMachineTest, check_drive_state_is_broadcasted_over_can)
         App_CanTx_GetPeriodicSignal_STATE(can_tx_interface));
 }
 
+// DIM_9
```
Comment (garethellis0) [inline boards/DIM/Test/Src/Test_StateMachine.cpp:192]: We should consistent about using `-` vs `_` in these specs eh? I think `-` better matches the specs markdown doc.

Comment (thekenu) [reply boards/DIM/Test/Src/Test_StateMachine.cpp:192]: Definitely unintentional, will fix

Comment (thekenu) [reply boards/DIM/Test/Src/Test_StateMachine.cpp]: Should be fixed now!

**Pull request merged**